vcn_gateways.tf

```diff
@@ -23,8 +23,52 @@ resource "oci_core_route_table" "ig" {
   freeform_tags = var.tags
 
   route_rules {
+    # * With this route table, Internet Gateway is always declared as the default gateway
     destination       = local.anywhere
     network_entity_id = oci_core_internet_gateway.ig[0].id
+    description       = "Terraformed - Auto-generated at Internet Gateway creation: Internet Gateway as default gateway"
+  }
+
+  dynamic "route_rules" {
+    # * filter var.internet_gateway_route_rules for routes with "drg" as destination
+    # * and steer traffic to the module created DRG
+    for_each = var.internet_gateway_route_rules != null ? { for k, v in var.internet_gateway_route_rules : k => v
+    if v.network_entity_id == "drg" } : {}
+
+    content {
+      destination       = route_rules.value.destination
+      destination_type  = route_rules.value.destination_type
+      network_entity_id = oci_core_drg.drg[0].id
+      description       = route_rules.value.description
+    }
+  }
+
+  dynamic "route_rules" {
+    # * filter var.internet_gateway_route_rules for routes with "internet_gateway" as destination
+    # * and steer traffic to the module created Internet Gateway
+    for_each = var.internet_gateway_route_rules != null ? { for k, v in var.internet_gateway_route_rules : k => v
+    if v.network_entity_id == "internet_gateway" } : {}
+
+    content {
+      destination       = route_rules.value.destination
+      destination_type  = route_rules.value.destination_type
+      network_entity_id = oci_core_internet_gateway.ig[0].id
+      description       = route_rules.value.description
+    }
+  }
+
+  dynamic "route_rules" {
```

Comment:
I don't quite follow the purpose of this 3rd route rule. You've already filtered IGW and DRG. What is this one used for then?

Reply:
The first two blocks generates a rule only if the following strings are matched: "drg" or "internet_gateway", matched with a gateway that is handled by the module.

This third rule is used to take any string as an input (the rules basically says if `v.network_entity_id` do NOT contains "drg" or "internet_gateway"), and is usefull for gateways that are not managed by the module and gives freedom choose how you pass the target ocid.

This enables the following scenarios:
- route with LPG as a target (despite vcn module do not support nor create LPG today)
- route to a gateway that is not create by the vcn module (let say customer have an existing DRG with VPN tunnels or FastConnect circuits, and needs to keep the same DRG).

Reply:
ah excellent! maybe update your comment - that would make it clearer

Reply:
Does it convey the idea? Formulation suggestions are welcome :-)

```hcl
# * filter var.internet_gateway_route_rules for generic routes
# * can take a string, a variable, a local, a data source, a resource, a module output ...
# * useful for gateways that are not managed by the module
```

Reply:
that works